src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs

```diff
@@ -210,16 +247,88 @@ impl HummockVersionExt for HummockVersion {
             .unwrap_or(0)
     }
 
+    fn init_with_parent_group(
+        &mut self,
+        parent_group_id: CompactionGroupId,
+        group_id: CompactionGroupId,
+        member_table_ids: &HashSet<StateTableId>,
+    ) -> (bool, Vec<(HummockSstableId, u64)>) {
+        let mut split_id_vers = vec![];
+        if parent_group_id == StaticCompactionGroupId::NewCompactionGroup as CompactionGroupId
+            || !self.levels.contains_key(&parent_group_id)
+        {
+            return (false, split_id_vers);
+        }
+        let (parent_levels, cur_levels) = unsafe {
+            let parent_levels = self.levels.get_mut(&parent_group_id).unwrap() as *mut Levels;
+            let cur_levels = self.levels.get_mut(&group_id).unwrap() as *mut Levels;
+            assert_ne!(parent_levels, cur_levels);
+            (&mut *parent_levels, &mut *cur_levels)
+        };
+        if let Some(ref mut l0) = parent_levels.l0 {
+            let mut insert_table_infos = vec![];
+            for sub_level in &mut l0.sub_levels {
+                for table_info in &mut sub_level.table_infos {
+                    if table_info
+                        .get_table_ids()
+                        .iter()
+                        .any(|table_id| member_table_ids.contains(table_id))
+                    {
+                        table_info.divide_version += 1;
+                        split_id_vers.push((table_info.get_id(), table_info.get_divide_version()));
+                        let mut branch_table_info = table_info.clone();
+                        table_info
+                            .table_ids
+                            .retain(|table_id| !member_table_ids.contains(table_id));
+                        branch_table_info
+                            .table_ids
+                            .retain(|table_id| member_table_ids.contains(table_id));
```

Comment:
> minor: these logics can be put into a function to avoid duplicate codes for handling L0 and other levels.

We will have to extract level_ids in #5890

Reply:
> minor: these logics can be put into a function to avoid duplicate codes for handling L0 and other levels.

I have used drain filter to refactor in #5890 , now I also use drain filter to refactor in #5888